server.go

```diff
@@ -1837,14 +1881,24 @@ func (s *server) ConnectToPeer(addr *lnwire.NetAddress, perm bool) error {
 	// connect to the target peer. If the we can't make the connection, or
 	// the crypto negotiation breaks down, then return an error to the
 	// caller.
+	return s.connectToPeer(addr, nil)
+
+}
+
+// connectToPeer establishes a connection to a remote peer. The done channel is
+// used to notify the caller of when the connection has been successfully
+// established.
+func (s *server) connectToPeer(addr *lnwire.NetAddress, done chan<- struct{}) error {
 	conn, err := brontide.Dial(s.identityPriv, addr, cfg.net.Dial)
 	if err != nil {
+		srvrLog.Errorf("Unable to connect to %v: %v", addr, err)
 		return err
 	}
 
-	// Once the connection has been made, we can notify the server of the
-	// new connection via our public endpoint, which will require the lock
-	// an add the peer to the server's internal state.
+	if done != nil {
+		done <- struct{}{}
```

Comment:
you can `close` the channel instead, which will never block :)

Reply:
Fixed.